pkg/services/login/authinfo.go

```diff
@@ -17,37 +18,102 @@ type AuthInfoService interface {
 }
 
 const (
+	// modules
 	SAMLAuthModule      = "auth.saml"
 	LDAPAuthModule      = "ldap"
 	AuthProxyAuthModule = "authproxy"
 	JWTModule           = "jwt"
 	RenderModule        = "render"
+	// OAuth provider modules
+	AzureADAuthModule    = "oauth_azuread"
+	GoogleAuthModule     = "oauth_google"
+	GitLabAuthModule     = "oauth_gitlab"
+	GithubAuthModule     = "oauth_github"
+	GenericOAuthModule   = "oauth_generic_oauth"
+	GrafanaComAuthModule = "oauth_grafana_com"
+	GrafanaNetAuthModule = "oauth_grafananet"
+	OktaAuthModule       = "oauth_okta"
+
+	// labels
+	SAMLLabel = "SAML"
+	LDAPLabel = "LDAP"
+	JWTLabel  = "JWT"
+	// OAuth provider labels
+	AuthProxtLabel    = "Auth Proxy"
+	AzureADLabel      = "AzureAD"
+	GoogleLabel       = "Google"
+	GenericOAuthLabel = "Generic OAuth"
+	GitLabLabel       = "GitLab"
+	GithubLabel       = "GitHub"
+	GrafanaComLabel   = "grafana.com"
+	OktaLabel         = "Okta"
 )
 
+// IsExternnalySynced is used to tell if the user is externallySynced or that we should the external sync
+// true means that the org role sync is handled by Grafana
+// Note: currently the users authinfo is overridden each time the user logs in
+// https://github.com/grafana/grafana/blob/4181acec72f76df7ad02badce13769bae4a1f840/pkg/services/login/authinfoservice/database/database.go#L61
+// this means that if the user has multiple auth providers and one of them is set to sync org roles
+// then the user will be synced from the provider they last logged in with
```

Comment:
nit: I think if only one provider is set to sync org roles, roles will only be synced when you log in with this provider. This comment makes it sound like they will be synced for all providers. Maybe something like:
```suggestion
// this means that if the user has multiple auth providers and one of them is set to sync org roles
// then IsExternallySynced will be true for this one provider and false for the others
```

Reply:
i see. let me look at it again and your suggestion

Reply:
yea this is more local to the function and is what happens. nice ! 👍